### PR TITLE
Github issue #1756, Fix pInheritanceInfo null ptr dereference

### PR DESCRIPTION
**Pull request by mark-lunarg**
Trying to execute an unrecorded secondary command buffer in an active renderpass resulted in a nullptr dereference. Added null check along with a warning for unrecorded command buffers.

We found no prohibition on executing empty command buffers, so if somebody knows differently we can upgrade the warning.

_CHECK IT OUT_:  MikeS found a cool and useful feature -- when in the 'Files changed' view, adding "?w=1" to the end of the URL in your browser will tell Github to ignore whitespace, making diffs with indent changes MUCH easier to understand.  It'll help with this one.

### CODE REVIEW

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -18962,6 +18962,39 @@ TEST_F(VkLayerTest, CommandQueueFlags) {
     }
 }
 
+TEST_F(VkLayerTest, ExecuteEmptySecondaryCB) {
+    TEST_DESCRIPTION("Attempt vkCmdExecuteCommands with a secondary command buffer that has not been recorded");
+
+    ASSERT_NO_FATAL_FAILURE(Init());
+    ASSERT_NO_FATAL_FAILURE(InitRenderTarget());
+    m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_WARNING_BIT_EXT, "is unrecorded and contains no commands");
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: This should be a hard error. The spec says, in VU for CmdExecuteCommands: 

> Any given element of pCommandBuffers must be in the _pending_ or _executable_ state.

A command buffer that has never been recorded, or has been reset, is in the _initial_ state.

(A command buffer in the _pending_ or _executable_ state but which happens to contain no commands is a different beast.)

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -3930,6 +3930,12 @@ static bool validateCommandBufferState(layer_data *dev_data, GLOBAL_CB_NODE *cb_
     if (CB_RECORDED != cb_state->state) {
         if (CB_INVALID == cb_state->state) {
             skip |= ReportInvalidCommandBuffer(dev_data, cb_state, call_source);
+        }
+        if (CB_NEW == cb_state->state) {
+            skip |= log_msg(dev_data->report_data, VK_DEBUG_REPORT_WARNING_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_COMMAND_BUFFER_EXT,
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: See complaint in test.

File: tests/layer_validation_tests.cpp (hunk 1 above)
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: We should probably align the state naming in CV with what the spec uses.

Comment (mark-lunarg) [inline tests/layer_validation_tests.cpp]: Sweet, will make the update in the test and check, and look at a follow on commit to update the state naming.

File: tests/layer_validation_tests.cpp [hunk 3]
```diff
@@ -18962,6 +18962,53 @@ TEST_F(VkLayerTest, CommandQueueFlags) {
     }
 }
 
+TEST_F(VkLayerTest, ExecuteEmptyCBs) {
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: Unrecorded?

**Pull request merged**